src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -0,0 +1,157 @@
+            seek_position (int): Position to start writing from
+            data (Union[bytes, str]): Contents to be written
+
+        Returns:
+            int: Number of bytes that were successfully written to the file
+        """
+        try:
+            with self.fs.open(path=path, mode=mode, encoding=encoding) as file_handle:
+                return file_handle.write(data)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def seek_file(
+        self,
+        path: str,
+        location: int = 0,
+        position: FileSeekPosition = FileSeekPosition.START,
+    ) -> int:
+        """Place the file pointer to the mentioned location in the file
+        relative to the position.
+
+        Args:
+            path (str): path of the file
+            location (int): Nth byte position. To be understood in relation to
+            the arg "position"
+            position (FileSeekPosition): from start of file, current location
+            or end of file
+
+        Returns:
+            int: file pointer location after seeking to the mentioned position
+        """
+        try:
+            with self.fs.open(path=path, mode="rb") as file_handle:
+                return file_handle.seek(location, position)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def make_dir(self, path: str, create_parents: bool = True):
+        """Create a directory.
+
+        Args:
+            path (str): Path of the directory to be created
+            create_parents (bool): Specify if parent directories to be created
+            if any of the nested directory does not exist
+        """
+        try:
+            self.fs.mkdir(path, create_parents=create_parents)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def path_exists(self, path: str) -> bool:
+        """Checks if a file/directory path exists.
+
+        Args:
+            path (str): File/directory path
+
+        Returns:
+            bool: If the file/directory  exists or not
+        """
+        try:
+            return self.fs.exists(path)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def list(self, path: str) -> list[str]:
+        """List the directory path.
+
+        Args:
+            path (str): Directory path
+
+        Returns:
+            List[str]: List of files / directories under the path
+        """
+        try:
+            return self.fs.ls(path)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def remove(self, path: str, recursive: bool = True):
+        """Removes a file or directory mentioned in path.
+
+        Args:
+            path (str): Path to the file / directory
+            recursive (bool): Whether the files and folders nested
+            under path are to be removed or not
+
+        Returns:
+        """
+        try:
+            return self.fs.rm(path, recursive=recursive)
```

Comment:
@gaya3-zipstack
1. What happens if a dir path is passed and recursive is False?
2. Also, what happens if a very large directory is triggered for delete?

Reply:
For (1), we get an error saying recusrive needs to be set to true if the folder to be deleted has nested files/folders under it.